conda/core/prefix_data.py

```diff
@@ -589,6 +591,18 @@ def _write_environment_state_file(self, state: dict[str, dict[str, str]]) -> Non
         env_vars_file = self.prefix_path / PREFIX_STATE_FILE
         env_vars_file.write_text(json.dumps(state, ensure_ascii=False))
 
+    def _ensure_no_reserved_env_vars(self, env_vars_names: Iterable[str]) -> None:
+        """
+        Ensure that the set of env_var_names does not contain any reserved env vars.
+        Will raise an OperationNotAllowed if a reserved env var is found.
+        """
+        for var in RESERVED_ENV_VARS:
+            if var in env_vars_names:
+                 raise OperationNotAllowed(
```

Comment:
This is pretty strict since if there are valid env vars to export in the set of `env_vars_names` they won't get exported. For example, if you have multiple variables definied in an `environment.yml` file.
 
Another approach is to:
* remove the reserved env var from the set of vars that are getting added to the config
* produce a log message/warning that the bad config is getting ignored.